### PR TITLE
Add multiple schema support

### PR DESCRIPTION
**Pull request by kkoehn**
Adds support for multiple schemas/schema-versions
Since the version is defined by the namespace, we have to store different schema-versions separately.
Validations are now executed by the Validator.
Importer, Exporter and Validator have an optional argument to supply the (expected) version of the schema to be used.

### CODE REVIEW

File: spec/proforma/importer_spec.rb [hunk 1]
```diff
@@ -125,5 +141,44 @@
...
+
+      context 'when exported_version is the same' do
+        let(:export_version) { expected_version }
+
+        it 'does not raise an error' do
+          expect { perform }.not_to raise_error
+        end
+      end
+
+      context 'when exported_version is different' do
```
Comment (MrSerth) [inline spec/proforma/importer_spec.rb:157]: I didn't get that test (and the corresponding importer). Why could the exported_version be different? From what?

**Review by MrSerth — APPROVED** [review]
Good solution 👍

File: spec/proforma/importer_spec.rb [hunk 2]
```diff
@@ -22,6 +23,20 @@
...
+    it 'does not assign expected_version' do
+      expect(importer.instance_variable_get(:@expected_version)).to be_nil
+    end
+
+    context 'with a specific expected_version' do
+      subject(:importer) { described_class.new(zip_file, expected_version) }
+
+      let(:expected_version) { '2.0' }
+
+      it 'does not assign expected_version' do
```
[the remark this answers is not in the document]
Comment (kkoehn) [reply spec/proforma/importer_spec.rb:36]: good catch

File: spec/proforma/validator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    end
+    let(:doc) { Nokogiri::XML(zip_content['task.xml'], &:noblanks) }
+    let(:validator) { described_class.new(doc) }
+    let(:export_version) {}
+
+    it { is_expected.to be_empty }
+
+    context 'when input is not a proforma xml' do
+      let(:doc) do
+        Nokogiri::XML('<root><aliens><alien><name>Alf</name></alien></aliens></root>')
```
[the remark this answers is not in the document]
Comment (kkoehn) [reply spec/proforma/validator_spec.rb:59]: this is shamelessly stolen from https://nokogiri.org/tutorials/parsing_an_html_xml_document.html

File: spec/proforma/importer_spec.rb (hunk 1 above)
Comment (kkoehn) [reply spec/proforma/importer_spec.rb]: I wanted to be able to restrict the version used for the import. This spec should check for that.

Comment (MrSerth) [reply spec/proforma/importer_spec.rb]: Ah, okay. Thanks

**Review by MrSerth — APPROVED** [review]

**Pull request merged**